common/src/env_vars.rs

```diff
@@ -0,0 +1,31 @@
+use std::{env::VarError, ffi::OsStr};
+
+pub enum EnvVar {
+    Agent,
+    Interface,
+    N3hPath,
+    N3hMode,
+    N3hWorkDir,
+    N3hBootstrapNode,
+    SimpleLoggerMute,
+    TargetPrefix,
+}
+
+impl AsRef<OsStr> for EnvVar {
```

Comment:
You don't have to actually implement `AsRef` for the `var` function to work. You can just map to `&'static str` values. That would still work.

Reply:
I was trying... but I couldn't figure it out, can you show me?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=enum%20Var%20%7B%0A%20%20%20%20A%2C%0A%20%20%20%20B%2C%0A%20%20%20%20C%0A%7D%0A%0Aimpl%20Var%20%7B%0A%20%20%20%20pub%20fn%20as_str(%26self)%20-%3E%20%26str%20%7B%0A%20%20%20%20%20%20%20%20match%20self%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20Var%3A%3AA%20%3D%3E%20%22A%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20Var%3A%3AB%20%3D%3E%20%22B%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20Var%3A%3AC%20%3D%3E%20%22C%22%2C%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20println!(%22%7B%3A%3F%7D%22%2C%20std%3A%3Aenv%3A%3Avar(Var%3A%3AA.as_str()))%3B%0A%7D